rplugin/python3/denite/scantree.py

```diff
@@ -0,0 +1,48 @@
+# ============================================================================
+# FILE: scantree.py
+# AUTHOR: Jorge Rodrigues <skeept at gmail.com>
+# License: MIT license
+# ============================================================================
+
+import sys
+from os.path import basename, join
+from os import walk
+
+try:
+    from os import scandir
+
+    def scantree(path_name, skip_list=None):
+        """if scandir is available use it as it is faster than os.walk"""
+
+        if skip_list is None:
+            skip_list = ['.git', '.hg']
+
+        for entry in scandir(path_name):
+            if entry.is_dir(follow_symlinks=False):
+                if basename(entry.path) not in skip_list:
+                    yield from scantree(entry.path)
+            else:
+                yield entry.path
+except ImportError:
+    def scantree(path_name, skip_list=None):
+        """scandir is not available use os.walk"""
+
+        if skip_list is None:
+            skip_list = ['.git', '.hg']
+
+        for root, dirn, files in walk(path_name):
+            if basename(root) in skip_list:
+                continue
+            for fn in files:
```

Comment:
It is a bit personal reason but I feel `fn` is a short form of `function` so `filename` is better in this case.